src/components/tl/ucp/tl_ucp.c

```diff
@@ -93,3 +101,29 @@ ucs_memory_type_t ucc_memtype_to_ucs[UCC_MEMORY_TYPE_LAST+1] = {
     [UCC_MEMORY_TYPE_ROCM_MANAGED] = UCS_MEMORY_TYPE_ROCM_MANAGED,
     [UCC_MEMORY_TYPE_UNKNOWN]      = UCS_MEMORY_TYPE_UNKNOWN
 };
+
+void ucc_tl_ucp_pre_register_mem(ucc_tl_ucp_team_t *team, void *addr,
+                                 size_t length, ucc_memory_type_t mem_type)
+{
+    void *base_address  = addr;
+    size_t alloc_length = length;
+    ucc_mem_attr_t mem_attr;
+    ucc_status_t status;
+
+    mem_attr.field_mask = UCC_MEM_ATTR_FIELD_BASE_ADDRESS |
+                          UCC_MEM_ATTR_FIELD_ALLOC_LENGTH;
+    status = ucc_mc_query(addr, length, &mem_attr);
+    if (status == UCC_OK) {
+        base_address = mem_attr.base_address;
+        alloc_length = mem_attr.alloc_length;
+    } else {
+       tl_warn(UCC_TL_TEAM_LIB(team), "failed to query base addr and len");
```

Comment:
when the failure can happen? do we need this warning? If this happens on each collective call wouldn't it clutter the output entirely?

Reply:
it possible that cuda API can return error from cuda mem component if something is really wrong.  I don't want to silently ignore this. it is readlly hard to say if it can happen each time.